twine/cli.py

```diff
@@ -37,6 +38,7 @@ def _registered_commands(
 
 def list_dependencies_and_versions() -> List[Tuple[str, str]]:
     return [
+        ("python", ".".join(map(str, sys.version_info))),
```

Comment:
I believe that `list_dependencies_and_versions` which is called via `twine --version` only focuses on module dependencies to `twine`. Python does not come under that umbrella IMO, so I think this change might be unnecessary.

On the other hand, if we are including `Python` version, it might be worthwhile to include other info like Python distribution, OS etc, like `flake8` does.

```
$ flake8 --version
3.8.1 (flake8-docstrings: 1.5.0, pydocstyle: 5.0.2, mccabe: 0.6.1, pycodestyle: 2.6.0, pyflakes: 2.2.0) CPython 3.8.2 on Darwin
```

Reply:
> I believe that `list_dependencies_and_versions` which is called via `twine --version` only focuses on module dependencies to `twine`. Python does not come under that umbrella IMO, so I think this change might be unnecessary.

This was a sentence in the style of "It's the way that it is because it's the way that it is" :)

The question is, do you think it's worthwhile information to be easily available when submitting bug reports?

> 
> On the other hand, if we are including `Python` version, it might be worthwhile to include other info like Python distribution, OS etc, like `flake8` does.
> 
> ```
> $ flake8 --version
> 3.8.1 (flake8-docstrings: 1.5.0, pydocstyle: 5.0.2, mccabe: 0.6.1, pycodestyle: 2.6.0, pyflakes: 2.2.0) CPython 3.8.2 on Darwin
> ```

If you want the information in this format, and see no reason why this change wouldn't be merged, let me know.

Reply:
> This was a sentence in the style of "It's the way that it is because it's the way that it is" :)

I won't interpret it that way. But  I do believe that it's worthwhile to discuss on the issue if a change like this to a user-facing output is good, especially for users who might depend on this output being a certain format.

> The question is, do you think it's worthwhile information to be easily available when submitting bug reports?

When creating a new issue, the template already asks for the python version via `python --version`.

> If you want the information in this format, and see no reason why this change wouldn't be merged, let me know.

It was more of a suggestion for improvement in the format, if the PR is considered, but that is up to the twine maintainers to decide if this will be merged or not.

Reply:
> When creating a new issue, the template already asks for the python version via `python --version`.

Yes, but in multi-Python setups, often the `twine` executable that's on the path uses a different Python version than the `python` executable on the path. 

> It was more of a suggestion for improvement in the format, if the PR is considered, but that is up to the twine maintainers to decide if this will be merged or not.

Cool, I'll wait for their take.

Reply:
Personally I think stuffing everything into `--version` is the wrong tact to take no matter what.

No one goes through the entire bug report template and rarely do folks ever provide `python --version`. (Just look at this PR which has no indication of how it was tested, no updates to tests, and no example of what the new output of `--version` would be.)

Flake8, and others are slowly adopting the approach of giving far more detail in a different, machine-parseable, format. See also `flake8 --bug-report`, `python -m requests.help`.

This also means we can include far more detail. It also allows for something like GitHub actions to take automated actions based on the issue body if we wanted that.

---

As to the semantics of `list_dependencies_and_versions` the name is clear, but that doesn't mean the name can't change. Regardless, I don't think this is the right direction forward and I certainly don't think a naming slap-fight is appropriate behaviour from either of you two.

Reply:
I understand. I'm sorry for unpleasantness.